internal/handlers/commonparams/extract_params.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commonparams
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// ExtractParams fill passed value structure with parameters from the document.
+// Parameters are extracted by the field name or by the `name` tag.
+//
+// Possible tags:
+// * `opt` - field is optional;
+// * `non-default` - field is unimplemented non-default;
+// * `unimplemented` - field is not implemented yet;
+// * `ignored` - field is ignored.
+func ExtractParams(doc *types.Document, command string, value any, l *zap.Logger) error {
+	rv := reflect.ValueOf(value)
+	if rv.Kind() != reflect.Ptr || rv.IsNil() {
+		return errors.New("unmarshal: value must be a non-nil pointer")
+	}
+
+	elem := rv.Elem()
+	if elem.Kind() != reflect.Struct {
+		return errors.New("unmarshal: value must be a struct pointer")
+	}
+
+	// Iterate over the fields of the struct.
+	for i := 0; i < elem.NumField(); i++ {
+		field := elem.Type().Field(i)
+
+		tag := field.Tag.Get("name")
+		if tag == "" {
+			tag = field.Name
+		}
+
+		var optional, nonDefault, unImplemented, ignored bool
+
+		for _, i := range strings.Split(tag, ",") {
+			switch i {
+			case "opt":
+				optional = true
+			case "non-default":
+				nonDefault = true
+			case "unimplemented":
+				unImplemented = true
```

Comment:
```suggestion
				unimplemented = true
```

It is a single word, no need to camelCase it